Makefile

```diff
@@ -4,37 +4,51 @@ PARAMS_FILE   := cloudformation/parameters.json
 EXCLUDE_DIRS  := .git|tests|cloudformation|requirements|$(STAGING_DIR)|$(BUILDS_DIR)
 PIP_COMMAND   := pip install -r
 
-.PHONY: init create-stack update-stack delete-stack describe-stack invoke test clean build _check-arn deploy
+.PHONY: init create-stack update-stack delete-stack describe-stack invoke test clean build update _check_config _check_stack_exists _is_user_authenticated
 
 # Read the cloudformation/parameters.json file for the ProjectName and EnvionmentName.
 # Use these to name the CloudFormation stack.
-PROJECT_NAME = $(shell cat $(PARAMS_FILE) | python -c 'import sys, json; j = [i for i in json.load(sys.stdin) if i["ParameterKey"]=="ProjectName"][0]["ParameterValue"]; print(j)')
-ENVIRONMENT_NAME = $(shell cat $(PARAMS_FILE) | python -c 'import sys, json; j = [i for i in json.load(sys.stdin) if i["ParameterKey"]=="EnvironmentName"][0]["ParameterValue"]; print(j)')
-STACK_NAME = $(PROJECT_NAME)-$(ENVIRONMENT_NAME)-stack
+PROJECT_NAME := $(shell cat $(PARAMS_FILE) | jq -r '.[] | select(.ParameterKey == "ProjectName") | .ParameterValue')
+ENVIRONMENT_NAME := $(shell cat $(PARAMS_FILE) | jq -r '.[] | select(.ParameterKey == "EnvironmentName") | .ParameterValue')
+S3_KEY_NAME := $(shell cat $(PARAMS_FILE) | jq -r '.[] | select(.ParameterKey == "S3DeploymentFileKey") | .ParameterValue')
+S3_BUCKET_NAME :=$(shell cat $(PARAMS_FILE) | jq -r '.[] | select(.ParameterKey == "S3DeploymentBucketName") | .ParameterValue')
+STACK_NAME := $(PROJECT_NAME)-$(ENVIRONMENT_NAME)
 
+_is_user_authenticated:
```

Comment:
What's the purpose of this?

Reply:
It basically just checks that the CLI is authenticated by running sts::get_caller - the idea is to fail quickly at the beginning for larger targets that might instead fail at the end with a less helpful error message. Not a critical change if you prefer the old method.